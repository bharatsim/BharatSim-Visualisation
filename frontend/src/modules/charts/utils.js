import { chartStyleConfig, getColor } from './chartStyleConfig';

function trasformDataForChart(csvData, xColumn, yColumns) {
  return {
    labels: csvData && csvData.data[xColumn],
    datasets:
      csvData &&
      yColumns.map((yColumn, index) => {
        return {
          ...chartStyleConfig,
          label: yColumn,
          borderColor: getColor(index),
          backgroundColor: getColor(index),
          data: csvData && csvData.data[yColumn],
        };
      }),
  };
}

function getYaxisNames(yColumns) {
  return yColumns.map((yColumn) => yColumn.name);
}

export { trasformDataForChart, getYaxisNames };
