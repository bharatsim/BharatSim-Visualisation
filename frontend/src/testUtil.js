// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent, within } from '@testing-library/react';

export function selectDropDownOption(container, dropDownId, optionId) {
  const dropDown = container.getByTestId(dropDownId);
  fireEvent.mouseDown(within(dropDown).getByRole('button'));
  const options = within(
    within(document.getElementById(`menu-${dropDownId}`)).getByRole('listbox'),
  );
  fireEvent.click(options.getByTestId(`${dropDownId}-${optionId}`));
}
