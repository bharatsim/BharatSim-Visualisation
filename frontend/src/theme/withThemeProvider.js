import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
