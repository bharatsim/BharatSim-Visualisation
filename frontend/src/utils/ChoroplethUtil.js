import L from 'leaflet';
import chroma from 'chroma-js';

function highlightOnClick(e) {
  const layer = e.target;

  layer.setStyle({
    weight: 2,
    dashArray: '',
    opacity: 1,
    fillOpacity: 0.7,
  });

  if (!L.Browser.ie && !L.Browser.opera && !L.Browser.edge) {
    layer.bringToFront();
  }
}

function resetHighlight(geoJson) {
  return (e) => geoJson.current.resetStyle(e.target);
}

function zoomToFeature(map) {
  return (e) => map.fitBounds(e.target.getBounds());
}

function geoJSONStyle(idDataMap, mapLayerIdName, maxOfMeasure, scale) {
  const colors = Object.keys(scale)
    .sort((v1, v2) => Number(v1) - Number(v2))
    .map((key) => scale[key]);
  const getColor = chroma.scale(colors);
  return (feature) => {
    const gradientPercentage = idDataMap[feature.properties[mapLayerIdName]] / maxOfMeasure;
    return {
      color: '#1f2021',
      weight: 1,
      fillOpacity: 0.5,
      opacity: 0.7,
      fillColor: getColor(gradientPercentage),
    };
  };
}

export { highlightOnClick, resetHighlight, geoJSONStyle, zoomToFeature };
