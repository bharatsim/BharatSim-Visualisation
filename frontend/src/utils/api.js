import { contentTypes, httpMethods } from '../constants/fetch';
import { url as serviceURL } from './url';
import { fetchData, uploadData } from './fetch';

function headerBuilder({ contentType }) {
  return {
    'content-type': contentType,
  };
}

function formDataBuilder(data) {
  const formData = new FormData();
  data.forEach(({ name, value }) => formData.append(name, value));
  return formData;
}

const api = {
  saveDashboard: async ({ widgets, layout, dashboardId, name, count }) => {
    return uploadData({
      url: serviceURL.DASHBOARD_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      data: JSON.stringify({
        dashboardData: { widgets, layout, dashboardId, name, count },
      }),
    });
  },

  addNewDashboard: async ({ name, projectId }) => {
    return uploadData({
      url: serviceURL.INSERT_DASHBOARD,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      data: JSON.stringify({
        dashboardData: { widgets: [], layout: [], name, count: 0, projectId },
      }),
      isCustomErrorHandler: true,
    });
  },

  getAllDashBoardByProjectId: async (projectId) =>
    fetchData({
      url: serviceURL.DASHBOARD_URL,
      query: { projectId, columns: ['name', '_id'] },
    }),

  getAllDashBoard: async () => fetchData({ url: serviceURL.DASHBOARD_URL }),

  uploadFileAndSchema: async ({ file, schema, dashboardId }) =>
    uploadData({
      url: serviceURL.DATA_SOURCES,
      data: formDataBuilder([
        { name: 'datafile', value: file },
        { name: 'schema', value: JSON.stringify(schema) },
        { name: 'dashboardId', value: dashboardId },
      ]),
      headers: headerBuilder({ contentType: contentTypes.FILE }),
    }),

  getCsvHeaders: async (dataSourceId) => fetchData({ url: serviceURL.getHeaderUrl(dataSourceId) }),

  getDatasources: async (dashboardId) =>
    fetchData({
      url: serviceURL.DATA_SOURCES,
      query: { dashboardId },
    }),

  getData: async (datasource, columns) =>
    fetchData({ url: serviceURL.getDataUrl(datasource), query: { columns } }),

  saveProject: async ({ id, ...data }) => {
    const requestObject = {
      url: serviceURL.PROJECT_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      isCustomErrorHandler: true,
    };
    if (id) {
      return uploadData({
        ...requestObject,
        data: JSON.stringify({ projectData: { ...data, id } }),
        method: httpMethods.PUT,
      });
    }
    return uploadData({
      ...requestObject,
      data: JSON.stringify({ projectData: { ...data } }),
      method: httpMethods.POST,
    });
  },

  getProjects: async () => {
    return fetchData({
      url: serviceURL.PROJECT_URL,
    });
  },
  getProject: async (id) => {
    return fetchData({
      url: serviceURL.getProjectUrl(id),
    });
  },
};

export { api };
