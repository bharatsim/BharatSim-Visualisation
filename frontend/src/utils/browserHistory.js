// eslint-disable-next-line import/no-mutable-exports
let BrowserHistory;

function initHistory(history) {
  BrowserHistory = history;
}

function navigateToPreviousPage() {
  BrowserHistory.goBack();
}

function navigateToHome() {
  BrowserHistory.push('/');
}

function reloadPage() {
  // eslint-disable-next-line no-restricted-globals
  location.reload();
}

export { initHistory, navigateToPreviousPage, reloadPage, BrowserHistory, navigateToHome };
