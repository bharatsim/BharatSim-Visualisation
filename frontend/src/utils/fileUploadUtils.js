import csvParser from 'papaparse';
import dataTypesMapping from '../constants/dataTypesMapping';
import csvParserConfig from '../config/csvParserConfig';
import { loaderStates } from '../hook/useLoader';

function getMessage(fileUploadStatus, fileName) {
  return {
    [loaderStates.ERROR]: `Error occurred while unloading ${fileName}`,
    [loaderStates.SUCCESS]: `${fileName} successfully uploaded`,
    [loaderStates.LOADING]: `uploading ${fileName}`,
  }[fileUploadStatus];
}
function createSchema(row) {
  return Object.keys(row).reduce((acc, element) => {
    acc[element] = dataTypesMapping[typeof row[element]];
    return acc;
  }, {});
}

function parseCsv(csvFile, onComplete, onError) {
  csvParser.parse(csvFile, {
    ...csvParserConfig,
    complete: onComplete,
    error: onError,
  });
}

function resetFileInput(fileInput) {
  // eslint-disable-next-line no-param-reassign
  fileInput.value = '';
  // eslint-disable-next-line no-param-reassign
  fileInput.files = null;
}

function createColumnForMTable(schema) {
  return Object.keys(schema).map((fieldName) => ({
    title: fieldName,
    field: fieldName,
    dataType: schema[fieldName],
  }));
}

export { getMessage, createSchema, parseCsv, resetFileInput, createColumnForMTable };
