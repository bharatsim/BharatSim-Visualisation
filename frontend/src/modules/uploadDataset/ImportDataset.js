import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Box from '@material-ui/core/Box';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import fileImportIcon from '../../assets/images/importFileIcon.svg';
import { createSchema, getFileExtension, parseCsv, parseJson } from '../../utils/fileUploadUtils';
import ErrorBar from '../../uiComponent/ErrorBar';
import { validateCSVFile, validateFile } from '../../utils/validators';
import { VALID_FILE_EXTENSIONS } from '../../constants/fileUpload';

const useStyles = makeStyles((theme) => {
  return {
    importDataContainer: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center',
      border: '1px dashed',
      borderRadius: theme.spacing(1),
      borderColor: theme.colors.primaryColorScale['200'],
    },
    textContainer: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      marginTop: theme.spacing(5),
    },
  };
});

function ImportDataset({ setFile, handleNext, setPreviewData, setErrorStep, setSchema }) {
  const classes = useStyles();
  const [error, setError] = useState();
  const previewLimit = 100;
  const fileParsers = { csv: readCsv, geojson: readJson };

  function onJsonParse(jsonData) {
    const { data, errors } = jsonData;
    if (errors.length > 0) {
      setError(
        'Failed to Import file due to parsing error. Please review the file and ensure that its a valid GEO JSON Data.',
      );
      setErrorStep(0);
      return;
    }
    setPreviewData(data);
    setSchema({});
    handleNext();
  }

  function onCsvParse(csvData) {
    const { data } = csvData;
    const validationError = validateCSVFile(csvData);
    if (validationError) {
      setError(validationError);
      setErrorStep(0);
      return;
    }
    setPreviewData(data);
    const schema = createSchema(data[0]);
    setSchema(schema);
    handleNext();
  }

  function readCsv(file) {
    parseCsv(file, previewLimit, onCsvParse);
  }

  function readJson(file) {
    parseJson(file, onJsonParse);
  }

  function parsefile(file) {
    fileParsers[getFileExtension(file)](file);
  }

  function handleFileImport(event) {
    setErrorStep(null);
    const selectedFile = event.target.files[0];
    const validationError = validateFile(selectedFile);
    if (validationError) {
      setError(validationError);
      setErrorStep(0);
      return;
    }
    parsefile(selectedFile);
    setFile(selectedFile);
  }

  return (
    <Box mb={8} mx={14}>
      <ErrorBar visible={!!error} message={error} />
      <Box py={8} mt={6} className={classes.importDataContainer}>
        <img src={fileImportIcon} alt="Import File" />
        <Box className={classes.textContainer}>
          <Typography>Drag your file here or </Typography>
          <Box ml={3}>
            <Button variant="outlined" component="label" size="small">
              Browse
              <input
                data-testid="file-input"
                type="file"
                accept={`.${VALID_FILE_EXTENSIONS.join(',.')}`}
                style={{ display: 'none' }}
                onChange={handleFileImport}
              />
            </Button>
          </Box>
        </Box>
      </Box>
    </Box>
  );
}

ImportDataset.propTypes = {
  setFile: PropTypes.func.isRequired,
  handleNext: PropTypes.func.isRequired,
  setPreviewData: PropTypes.func.isRequired,
  setErrorStep: PropTypes.func.isRequired,
  setSchema: PropTypes.func.isRequired,
};

export default ImportDataset;
