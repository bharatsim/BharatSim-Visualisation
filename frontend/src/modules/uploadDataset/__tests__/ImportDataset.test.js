import React from 'react';
import { fireEvent } from '@testing-library/dom';
import withThemeProvider from '../../../theme/withThemeProvider';
import ImportDataset from '../ImportDataset';
import * as fileUtils from '../../../utils/fileUploadUtils';
import { renderWithRedux as render } from '../../../testUtil';

jest.spyOn(fileUtils, 'parseCsv').mockImplementation((csvFile, previewLimit, onComplete) => {
  const data = {
    data: [{ col1: 'row1', col2: 1 }],
    errors: [],
    meta: { fields: ['col1', 'col2'] },
  };
  onComplete(data);
});

jest.spyOn(fileUtils, 'parseJson').mockImplementation((jsonFile, onComplete) => {
  const data = { data: [{ col1: 'row1', col2: 1 }], errors: [] };
  onComplete(data);
});

describe('Import Dataset', () => {
  const Component = withThemeProvider(ImportDataset);
  let setFileMock;
  let handleNextMock;
  let setPreviewDataMock;
  let setErrorStepMock;
  let setSchemaMock;
  beforeEach(() => {
    setFileMock = jest.fn();
    handleNextMock = jest.fn();
    setPreviewDataMock = jest.fn();
    setErrorStepMock = jest.fn();
    setSchemaMock = jest.fn();
  });

  it('should match snapshot for import dataset component', () => {
    const { container } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
        setSchema={setSchemaMock}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should import and set csv file ', () => {
    const { getByTestId } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
        setSchema={setSchemaMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [{ name: 'test.csv', size: '10' }] },
    });

    expect(setFileMock).toHaveBeenCalledWith({ name: 'test.csv', size: '10' });
  });
  it('should import and parse csv file ', () => {
    const { getByTestId } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
        setSchema={setSchemaMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [{ name: 'test.csv', size: '10' }] },
    });

    expect(fileUtils.parseCsv).toHaveBeenCalledWith(
      { name: 'test.csv', size: '10' },
      100,
      expect.any(Function, () => {}),
    );
  });
  it('should import and parse json file', () => {
    const { getByTestId } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
        setSchema={setSchemaMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [{ name: 'test.json', size: '10' }] },
    });

    expect(fileUtils.parseJson).toHaveBeenCalledWith(
      { name: 'test.json', size: '10' },
      expect.any(Function, () => {}),
    );
  });

  it('should create and set schema for csv files', () => {
    const { getByTestId } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
        setSchema={setSchemaMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [{ name: 'test.csv', size: '10' }] },
    });

    expect(setSchemaMock).toHaveBeenCalledWith({
      col1: 'String',
      col2: 'Number',
    });
  });
  it('should set preview data for csv file', () => {
    const { getByTestId } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
        setSchema={setSchemaMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [{ name: 'test.csv', size: '10' }] },
    });

    expect(setPreviewDataMock).toHaveBeenCalledWith([{ col1: 'row1', col2: 1 }]);
  });
  it('should set preview data for json file', () => {
    const { getByTestId } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
        setSchema={setSchemaMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [{ name: 'test.json', size: '10' }] },
    });

    expect(setPreviewDataMock).toHaveBeenCalledWith([{ col1: 'row1', col2: 1 }]);
  });
  it('should set error for any parsing error in given csv file', () => {
    jest.spyOn(fileUtils, 'parseCsv').mockImplementation((csvFile, previewLimit, onComplete) => {
      const data = {
        data: [{ col1: 'row1', col2: 1 }],
        errors: ['error'],
        meta: { fields: ['col1', 'col2'] },
      };
      onComplete(data);
    });
    const { getByTestId, queryByText } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
        setSchema={setSchemaMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [{ name: 'test.csv', size: '10' }] },
    });

    expect(
      queryByText(
        'Failed to Import file due to parsing error. Please review the file and ensure that its a valid CSV file.',
      ),
    ).toBeInTheDocument();
    expect(setErrorStepMock).toHaveBeenCalledWith(0);
  });
  it('should set error for any parsing error in given json file', () => {
    jest.spyOn(fileUtils, 'parseJson').mockImplementation((csvFile, onComplete) => {
      const data = { data: [{ col1: 'row1', col2: 1 }], errors: ['error'] };
      onComplete(data);
    });
    const { getByTestId, queryByText } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
        setSchema={setSchemaMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [{ name: 'test.json', size: '10' }] },
    });

    expect(
      queryByText(
        'Failed to Import file due to parsing error. Please review the file and ensure that its a valid JSON Data.',
      ),
    ).toBeInTheDocument();
    expect(setErrorStepMock).toHaveBeenCalledWith(0);
  });
  it('should set validation error for given file if size exceed limit of 300mb', () => {
    jest.spyOn(fileUtils, 'parseCsv').mockImplementation((csvFile, previewLimit, onComplete) => {
      const data = {
        data: [{ col1: 'row1', col2: 1 }],
        errors: [],
        meta: { fields: ['col1', 'col2'] },
      };
      onComplete(data);
    });
    const { getByTestId, queryByText } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
        setSchema={setSchemaMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [{ name: 'test.csv', size: '120843092842123' }] },
    });

    expect(
      queryByText('Failed to Import file, size exceeds the limit of 300MB'),
    ).toBeInTheDocument();
    expect(setErrorStepMock).toHaveBeenCalledWith(0);
  });
  it('should set validation error for given file if type if not supported', () => {
    jest.spyOn(fileUtils, 'parseCsv').mockImplementation((csvFile, previewLimit, onComplete) => {
      const data = {
        data: [{ col1: 'row1', col2: 1 }],
        errors: [],
        meta: { fields: ['col1', 'col2'] },
      };
      onComplete(data);
    });
    const { getByTestId, queryByText } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
        setSchema={setSchemaMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [{ name: 'file.txt', size: '120843092842123' }] },
    });

    expect(queryByText('Failed to Import file, the format is not supported')).toBeInTheDocument();
    expect(setErrorStepMock).toHaveBeenCalledWith(0);
  });
  it('should set validation error for given file if csv file has _id as column', () => {
    jest.spyOn(fileUtils, 'parseCsv').mockImplementation((csvFile, previewLimit, onComplete) => {
      const data = { data: [{ _id: 'row1', col2: 1 }], errors: [], meta: { fields: ['_id'] } };
      onComplete(data);
    });
    const { getByTestId, queryByText } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
        setSchema={setSchemaMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [{ name: 'file.csv', size: '120843092' }] },
    });

    expect(
      queryByText('Failed to Import file due to wrong column name. Please change _id column name'),
    ).toBeInTheDocument();
    expect(setErrorStepMock).toHaveBeenCalledWith(0);
  });
});
