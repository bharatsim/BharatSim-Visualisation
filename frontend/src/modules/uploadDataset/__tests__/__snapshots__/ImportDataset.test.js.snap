// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import Dataset should match snapshot for import dataset component 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <div
      class="MuiBox-root MuiBox-root-9 makeStyles-hidden-6"
    >
      <div
        class="MuiBox-root MuiBox-root-10"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 makeStyles-errorTitle-4 MuiTypography-colorError"
        >
          Error:Â 
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 makeStyles-errorMessage-5 MuiTypography-colorError"
        />
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-11 makeStyles-importDataContainer-1"
    >
      <img
        alt="Import File"
        src="importFileIcon.svg"
      />
      <div
        class="MuiBox-root MuiBox-root-12 makeStyles-textContainer-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Drag your file here or 
        </p>
        <div
          class="MuiBox-root MuiBox-root-13"
        >
          <label
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Browse
              <input
                accept=".geojson,.csv"
                data-testid="file-input"
                style="display: none;"
                type="file"
              />
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
